stdlib/REPL/src/LineEdit.jl

```diff
@@ -1560,7 +1560,7 @@ function add_nested_key!(keymap::Dict, key::Union{String, Char}, value; override
         if y === nothing
             keymap[c] = value
             break
-        elseif !(c in keys(keymap) && isa(keymap[c], Dict))
+        elseif !((c in keys(keymap))::Bool && isa(keymap[c], Dict))
```

Comment:
I'm not at all familiar with the invalidation business, but judging from the recurrent patterns: the negation acts on the two expressions whose final value is passed to `!`, so are you sure you don't need

```suggestion
        elseif !((c in keys(keymap) && isa(keymap[c], Dict))::Bool)
```

Reply:
I checked with Cthulu and the problem was that `c in keys(keymap)` was inferred as `Any`.